pkg/operation/botanist/secrets.go

```diff
@@ -552,6 +552,16 @@ func (b *Botanist) generateSecrets() ([]interface{}, error) {
 		fmt.Sprintf("kubernetes.default.svc.%s", gardenv1beta1.DefaultDomain),
 		b.Shoot.InternalClusterDomain,
 	}
+
+	etcdCertDNSNames := []string{
```

Comment:
It's better to use a headless kubernetes service and the API server to talk to `etcd-{role}-0.etcd-{role}`

Reply:
Why do you suggest that we need the etcd client as a headless service? For peer communication endpoints, it makes sense because the communication has to be directed to a particular instance instead of any instances. For client endpoints, we have the liberty of directing the request to any of the instances (though we have just one).